projects/step-core/src/lib/modules/table/shared/table-remote-data-source.ts

```diff
@@ -41,43 +44,31 @@ export class TableRequest {
 
 const convertTableRequest = (req: TableRequest): TableRequestData => {
   const result: TableRequestData = {
-    draw: 1,
-    columns: [],
     start: req.start || 0,
     length: req.length || 10,
-    order: [],
-    search: {
-      value: '',
-      regex: false,
-    },
-    filter: req.filter,
   };
 
-  result.columns = req.columns.map((name) => {
-    const searchColumn = (req.searchBy || []).find((s) => s.column === name);
-    return searchColumn
-      ? {
-          name,
-          searchable: true,
-          search: {
-            value: searchColumn.search,
-            regex: searchColumn.regex,
-          },
-        }
-      : { name };
-  });
-
-  let orderColumnIndex = 0; //if not specified otherwise, the first column is sorted ascendingly
-  if (!!req.orderBy && req.columns.indexOf(req.orderBy.column) >= 0) {
-    orderColumnIndex = req.columns.indexOf(req.orderBy.column);
+  if (req.searchBy && req.searchBy.length > 0) {
+    result.filters = req.searchBy.map(({ column: field, search: value, regex: isRegex }) => ({
+      field,
+      value,
+      isRegex,
+    }));
   }
-  result.columns[orderColumnIndex].orderable = true;
-  result.order = [
-    {
-      column: orderColumnIndex,
-      dir: !!req.orderBy?.order ? req.orderBy.order : 'asc',
-    },
-  ];
+
+  result.sort = req.orderBy
+    ? {
+        field: req.orderBy.column,
+        direction: req.orderBy.order === 'asc' ? SortDirection.ASCENDING : SortDirection.DESCENDING,
+      }
+    : {
+        field: req.columns[0],
+        direction: SortDirection.ASCENDING,
+      };
+
+  // if (req.filter) {
+  // todo vad: for now it is not clear how to proceed queries like filter: "not(resourceType=attachment)'"
+  // }
```

Comment:
@dvladir this can no be done like this, I had not had time to update the code, can you quickly add this?

 {"oql":"not(resourceType=attachment)"}